src/maybe_nan/mod.rs

```diff
@@ -241,6 +241,15 @@ where
         A: 'a,
         F: FnMut(B, &'a A::NotNan) -> B;
 
+    /// Traverse the non-NaN elements and their indices and apply a fold,
+    /// returning the resulting value.
+    ///
+    /// Elements are visited in arbitrary order.
+    fn indexed_fold_skipnan<'a, F, B>(&'a self, init: B, f: F) -> B
```

Comment:
BTW, I wonder if it is better if we introduce `indexed_iter_skipnan` instead of `indexed_fold_skipnan` so that it can be used for different purposes, i.e. `indexed_inter_skipnan().fold()` is one of the usecases.

Reply:
I wouldn't mind adding `indexed_iter_skipnan()` in addition to `indexed_fold_skipnan()`, but the semantics of `indexed_iter_skipnan().fold()` and `indexed_fold_skipnan()` are a bit different. (I'd assume that `indexed_iter_skipnan()` would iterate in standard order to be consistent with `indexed_iter()`, while I wouldn't assume that for `indexed_fold_skipnan()`.)

Reply:
Ah ok, I forgot about the order semantic, in that case, perhaps `indexed_iter_skipnan` can be added later when it has actual usecase, wdyt? is there anything you want me to look into before merging this?